body.tex

```diff
@@ -1,166 +1,464 @@
+The RDS databases can be backed up into snapshots as well as exported to downloadable files on S3.
+\subsection{HTCondor}
+\href{https://research.cs.wisc.edu/htcondor}{HTCondor}\citep{condor-practice} provides distributed job parallelization and is a powerful batch system for high throughput computing (HTC).
+A HTCondor pool is a collection of compute resources, and HTCondor matches job requests with available resources following their job requirements.
+The \texttt{condor\_annex} module allows HTCondor deployment on cloud resources. Instances allocated by \texttt{condor\_annex} are added to a shared resource pool on which jobs added in the job queue are scheduled on. Unused compute resources are automatically deallocated after some set time spent idling.
+HTCondor has been succesfully used by \href{https://aws.amazon.com/blogs/aws/experiment-that-discovered-the-higgs-boson-uses-aws-to-probe-nature}{HEP Cloud project}, scaling up to 60,000 cores in multi-region deployement on AWS, and by \href{https://www.linkedin.com/pulse/using-50k-gpus-across-multiple-clouds-icecube-science-igor-sfiligoi}{IceCube experiment}, where they scaled up to 51,000 GPUs across multiple regions and multiple cloud providers (AWS, Azure, and Google Cloud).
+\subsection{Pegasus}
+\href{https://pegasus.isi.edu/}{Pegasus}\citep{deelman-fgcs-2015} is a workflow management system built on top of HTCondor.
+Popular in large-scale scientific computing, it provides command line and API interfaces for scientists to write abstract workflow independent of the underlying computing infrastructure.
+Pegasus workflows are expressed as Directed Acyclic Graphs (DAGs) where jobs are the nodes and job dependencies are represented by the edges.
+Pegasus utilizes HTCondor DAGMan as its execution engine, but also supports various execution environments and data management strategies.
+\subsection{LSST Software Stack}
+The \href{https://pipelines.lsst.io/}{LSST Science Pipelines Software Stack} is a collection of data processing codes for optical and near-infrared astronomy, and is used for DRP \citep{2018PASJ...70S...5B}.
+Besides image processing and astronomical catalog analysis, it includes a middleware layer for data access, task definition and execution.
+The LSST Data Butler is a framework that abstracts the underlying data persistence and implementations from Science Pipelines algorithms, and a data repository is a set of datasets under control of the Data Butler \citep{2018arXiv181208085J}.
+The Data Butler concepts can be found in \citeds{LDM-463}.
+Recently, the middleware layer is ungergoing a major refactoring, with the new middleware known as the Generation 3 Middleware ("Gen3" Middleware; \citeds{DMTN-056}).
```

Comment:
"The middleware layer is in the course of a major redesign and refactoring, ..."

Reply:
Rephrased. Thanks.